Fix #1833: Rename package to org.oppia.android (part 2/2)

BenHenning opened the pull request:
Fix #1833.

This PR renames the Android package to org.oppia.android. A few things to note:
1. This keeps org.oppia.app (it's now org.oppia.android.app) since we have separate modules. In the future, we may restructure things after #59 is completed to have more top-level directories than the current module-based structure imposes.
2. Because of (1), all R and databinding references are now relative to org.oppia.android instead of .app
3. There are some real fixes to module references & other test updates to make Bazel work again (these tests are probably broken on develop, too)
4. The Bazel files have been reformatted due to my Android Studio plugin automatically formatting them
5. Multidex was switched to native despite this breaking pre-L devices (see #1875 for specifics)
6. The Firebase project has been updated, but I can't tell if it's successfully connecting to Firebase (/cc @Sarthak2601 as FYI)

Note that this is a continuation of #1879. This PR will be merged immediately after #1879 goes in & CI checks are passing (which means the codebase will be intermittently broken for a short period of time--submissions will be disabled during that point).

BenHenning committed:
Move all source files to new org/oppia/android directory structure. No
actual package or code references are updated yet.

BenHenning committed:
Update Kotlin & XML references to match new package structure.

BenHenning committed:
Migrate codepaths for Bazel builds to new package structure.

BenHenning committed:
Fix package references in Gradle files for new package structure.

BenHenning committed:
Update package & pathing references to new structure for proto files.

BenHenning committed:
Rename Android package to org.oppia.android.

This includes introducing a new developer Firebase project. It also
fixes references to Android resource R file imports & databinding since
they both exist at the package level rather than the app module level.

BenHenning committed:
Fix Bazel builds for new package structure.

This required some changes to test setup/arrangement in Bazel, and some
existing issues needed to be fixed (missing dependencies for work
manager, switching multidex, one renamed test, incorrect module
references in tests, and files now referencing resources needing to be
moved).

This includes some Bazel file reformatting that happened automatically
by my Bazel file plugin.

BenHenning committed:
Merge branch 'develop' into introduce-new-package-structure

Note that this intentionally keeps the merge artifacts in since they
will be resolved en masse with find-and-replace commands that will
undergo manual verification.

Conflicts:
	app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsViewModel.kt
	app/src/main/java/org/oppia/android/app/administratorcontrols/administratorcontrolsitemviewmodel/AdministratorControlsDownloadPermissionsViewModel.kt
	app/src/main/java/org/oppia/android/app/application/ApplicationInjector.kt
	app/src/main/java/org/oppia/android/app/completedstorylist/CompletedStoryListViewModel.kt
	app/src/main/java/org/oppia/android/app/drawer/NavigationDrawerFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/home/HomeFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/ongoingtopiclist/OngoingTopicListViewModel.kt
	app/src/main/java/org/oppia/android/app/options/OptionControlsViewModel.kt
	app/src/main/java/org/oppia/android/app/options/OptionsFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/player/audio/AudioFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivityPresenter.kt
	app/src/main/java/org/oppia/android/app/player/exploration/ExplorationManagerFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/player/exploration/HintsAndSolutionExplorationManagerFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt
	app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt
	app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt
	app/src/main/java/org/oppia/android/app/profile/PinPasswordViewModel.kt
	app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/profile/ProfileChooserViewModel.kt
	app/src/main/java/org/oppia/android/app/profile/ResetPinDialogFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/profileprogress/ProfilePictureActivityPresenter.kt
	app/src/main/java/org/oppia/android/app/profileprogress/ProfileProgressViewModel.kt
	app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditActivityPresenter.kt
	app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditViewModel.kt
	app/src/main/java/org/oppia/android/app/settings/profile/ProfileListViewModel.kt
	app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameActivityPresenter.kt
	app/src/main/java/org/oppia/android/app/settings/profile/ProfileResetPinActivityPresenter.kt
	app/src/main/java/org/oppia/android/app/splash/SplashActivityPresenter.kt
	app/src/main/java/org/oppia/android/app/story/StoryViewModel.kt
	app/src/main/java/org/oppia/android/app/topic/TopicViewModel.kt
	app/src/main/java/org/oppia/android/app/topic/info/TopicInfoFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/topic/practice/TopicPracticeViewModel.kt
	app/src/main/java/org/oppia/android/app/topic/questionplayer/HintsAndSolutionQuestionManagerFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/topic/revision/TopicRevisionViewModel.kt
	app/src/main/java/org/oppia/android/app/walkthrough/end/WalkthroughFinalFragmentPresenter.kt
	app/src/main/java/org/oppia/android/app/walkthrough/welcome/WalkthroughWelcomeFragmentPresenter.kt
	app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt
	app/src/sharedTest/java/org/oppia/android/app/profile/AddProfileActivityTest.kt
	app/src/sharedTest/java/org/oppia/android/app/profile/AdminAuthActivityTest.kt
	app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt
	app/src/sharedTest/java/org/oppia/android/app/profile/PinPasswordActivityTest.kt
	app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt
	app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt
	app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameActivityTest.kt
	app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileResetPinActivityTest.kt
	app/src/test/java/org/oppia/android/app/home/HomeActivityLocalTest.kt
	app/src/test/java/org/oppia/android/app/player/exploration/ExplorationActivityLocalTest.kt
	app/src/test/java/org/oppia/android/app/player/state/StateFragmentLocalTest.kt
	app/src/test/java/org/oppia/android/app/testing/CompletedStoryListSpanTest.kt
	app/src/test/java/org/oppia/android/app/testing/OngoingTopicListSpanTest.kt
	app/src/test/java/org/oppia/android/app/testing/ProfileChooserSpanTest.kt
	app/src/test/java/org/oppia/android/app/testing/ProfileProgressSpanCount.kt
	app/src/test/java/org/oppia/android/app/testing/TopicRevisionSpanTest.kt
	app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt
	app/src/test/java/org/oppia/android/app/testing/options/DefaultAudioFragmentTest.kt
	app/src/test/java/org/oppia/android/app/testing/options/OptionsFragmentTest.kt
	app/src/test/java/org/oppia/android/app/testing/options/ReadingTextSizeFragmentTest.kt
	app/src/test/java/org/oppia/android/app/testing/player/split/PlayerSplitScreenTesting.kt
	app/src/test/java/org/oppia/android/app/testing/player/state/StateFragmentAccessibilityTest.kt
	app/src/test/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityLocalTest.kt
	data/src/test/java/org/oppia/android/data/persistence/PersistentCacheStoreTest.kt
	domain/src/main/java/org/oppia/android/domain/audio/CellularAudioDialogController.kt
	domain/src/main/java/org/oppia/android/domain/exploration/ExplorationDataController.kt
	domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt
	domain/src/main/java/org/oppia/android/domain/onboarding/AppStartupStateController.kt
	domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt
	domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentProgressController.kt
	domain/src/main/java/org/oppia/android/domain/question/QuestionTrainingController.kt
	domain/src/main/java/org/oppia/android/domain/topic/StoryProgressController.kt
	domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt
	domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt
	domain/src/test/java/org/oppia/android/domain/audio/CellularAudioDialogControllerTest.kt
	domain/src/test/java/org/oppia/android/domain/exploration/ExplorationDataControllerTest.kt
	domain/src/test/java/org/oppia/android/domain/exploration/ExplorationProgressControllerTest.kt
	domain/src/test/java/org/oppia/android/domain/onboarding/AppStartupStateControllerTest.kt
	domain/src/test/java/org/oppia/android/domain/oppialogger/analytics/AnalyticsControllerTest.kt
	domain/src/test/java/org/oppia/android/domain/oppialogger/exceptions/ExceptionsControllerTest.kt
	domain/src/test/java/org/oppia/android/domain/oppialogger/exceptions/UncaughtExceptionLoggerStartupListenerTest.kt
	domain/src/test/java/org/oppia/android/domain/profile/ProfileManagementControllerTest.kt
	domain/src/test/java/org/oppia/android/domain/question/QuestionAssessmentProgressControllerTest.kt
	domain/src/test/java/org/oppia/android/domain/question/QuestionTrainingControllerTest.kt
	domain/src/test/java/org/oppia/android/domain/topic/StoryProgressControllerTest.kt
	domain/src/test/java/org/oppia/android/domain/topic/StoryProgressTestHelperTest.kt
	domain/src/test/java/org/oppia/android/domain/topic/TopicControllerTest.kt
	domain/src/test/java/org/oppia/android/domain/topic/TopicListControllerTest.kt
	testing/src/main/java/org/oppia/android/testing/profile/ProfileTestHelper.kt
	testing/src/test/java/org/oppia/android/testing/profile/ProfileTestHelperTest.kt
	utility/src/test/java/org/oppia/android/util/data/DataProvidersTest.kt

BenHenning committed:
Resolve merge conflicts.

This won't compile due to duplicate imports which requires a ktlint
clean-up. That will happen in a subsequent commit since it will likely
pick up unrelated changes.

BenHenning committed:
Remove some merge conflict markers that were missed.

BenHenning committed:
Remove redundant commits & other ktlint updates.

BenHenning committed:
Fix incorrect references to new testing package location.

BenHenning committed:
Post-testing module cleanups via ktlint.

BenHenning committed:
More post-merge fixes.

Fix new references to R and databinding. Move EditTextInputAction to
correct package.

BenHenning committed:
Move some missed files that only caused breakages in Bazel.

BenHenning commented:
@rt4914 @anandwana001 @vinitamurthi I'm concerned about this PR. GitHub is not recognizing these are moved files, which means it's going to destroy history (not to mention, this PR is impossibly long to review). One alternative that we have is that I could split this into two PRs:
1. One that just moves all the files without changing any (which should definitely appear as moves in GitHub & retain history)
2. One that updates all the package references & Kotlin imports

While that will reduce the number of lines to review by an order of magnitude (and make it trivial since the lines changed are mostly just imports), it means that the repository is fully broken between (1) & (2). We also have to create (2) after (1) is merged otherwise we run into the same problem that we have now.

Any thoughts or suggestions?

rt4914 commented:
> @rt4914 @anandwana001 @vinitamurthi I'm concerned about this PR. GitHub is not recognizing these are moved files, which means it's going to destroy history (not to mention, this PR is impossibly long to review). One alternative that we have is that I could split this into two PRs:
> 
> 1. One that just moves all the files without changing any (which should definitely appear as moves in GitHub & retain history)
> 2. One that updates all the package references & Kotlin imports
> 
> While that will reduce the number of lines to review by an order of magnitude (and make it trivial since the lines changed are mostly just imports), it means that the repository is fully broken between (1) & (2). We also have to create (2) after (1) is merged otherwise we run into the same problem that we have now.
> 
> Any thoughts or suggestions?

@BenHenning We can decide a particular time and during that time we can finish part 1 and 2 so that even if it breaks it will break for a very less time.
For example: You can create part (1) in your morning time, I can review it at that moment directly and then you can submit second PR and i can review that too. If second PR can be done within your office hours, I can review it in that time.

I am thinking of you time because most of our contributors are from India.

BenHenning committed:
Move all files to their correct new location.

This moves all Kotlin source files to be under an org.oppia.android
structure, but does not actually change references. This is done in two
steps to try and preserve history, and to simplify the code review.

THIS WILL BREAK THE CODEBASE WHEN CHECKED IN.

BenHenning commented:
Sounds good--thanks @rt4914. As discussed during the team meeting, we'll do this today & split it into two PRs.

BenHenning committed:
Merge branch 'move-files-to-prepare-for-new-app-package-structure' into introduce-new-package-structure

Conflicts:
	app/src/main/java/org/oppia/android/app/application/ApplicationComponent.kt
	domain/src/main/java/org/oppia/android/domain/classify/rules/RuleQualifiers.kt

BenHenning commented:
Aha, it looks like changing the base does work. I'll send both PRs into review now so that we can make sure all approvals are in order before attempting submission.

BenHenning commented:
@rt4914 @anandwana001 PTAL.

rt4914 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -1,109 +1,108 @@
-    <meta-data android:name="firebase_analytics_collection_deactivated" android:value="true" />
-    <meta-data android:name="firebase_crashlytics_collection_enabled" android:value="false" />
+    <meta-data android:name="firebase_analytics_collection_deactivated" android:value="false" />
+    <meta-data android:name="firebase_crashlytics_collection_enabled" android:value="true" />
```
Why are these values changed?

BenHenning replied:
Great catch--this should be disabled. I'll actually file an issue to add a test to verify this so that this doesn't get accidentally regressed in the future.

#1903 for that.

BenHenning replied:
Fixed in https://github.com/oppia/oppia-android/pull/1876/commits/5239cfe0bbf8f79c2bed7a489ab628c15ebac556.

BenHenning committed:
Fix missed files from the migration after merging.

rt4914 reviewed (changes requested):
Reviewed first 300 files. 
Leaving two comments so that @BenHenning can update in parallel.

Have a look at this empty package. I think this should be removed. Right?

<img width="248" alt="Screenshot 2020-09-25 at 1 20 55 AM" src="https://user-images.githubusercontent.com/9396084/94193068-7b621e00-fecd-11ea-8fae-d56d4267e4a5.png">

rt4914 reviewed (changes requested):
Reviewed fully just waiting for reply on above comment.

rt4914 reviewed (approved):
Approving this as don't want to block this at this stage.
@BenHenning Just reply to comments and feel free to merge PR.

BenHenning committed:
Disable Firebase data collection.

BenHenning commented:
@rt4914 regarding the empty folders, that's just a local view that you have. Directories aren't tracked by git, just files. I suspect we may all have some empty folders after updating to the changes in this PR, and I'm not sure that I can do anything about that.

Otherwise, I think both of your comments have been addressed. PTAL and verify.

rt4914 reviewed (approved):
LGTM, thanks.

BenHenning committed:
Update BUILD.bazel

Capitalize TODO comment for consistency with other TODO comments.

BenHenning commented:
Thanks @rt4914. @anandwana001 I'm going to go ahead and merge this to try and get it in before SLoP folks start ramping up. Please let me know afterwards if you have any concerns with the PR and I'll be happy to address them in a subsequent PR.

rt4914 commented:
> Thanks @rt4914. @anandwana001 I'm going to go ahead and merge this to try and get it in before SLoP folks start ramping up. Please let me know afterwards if you have any concerns with the PR and I'll be happy to address them in a subsequent PR.

Sounds good. Thanks for wrapping up this quickly.

BenHenning committed:
Merge branch 'develop' into move-files-to-prepare-for-new-app-package-structure

BenHenning committed:
Merge branch 'move-files-to-prepare-for-new-app-package-structure' into introduce-new-package-structure

BenHenning commented:
#1879 has been merged & the history of this branch has been updated. Diff looks clean against develop, so just waiting for CI checks to pass before merging. Hopefully the history is retained post-merge.

BenHenning commented:
Looks like history wasn't retained: https://github.com/oppia/oppia-android/commits/develop/app/src/main/java/org/oppia/android/app/activity/ActivityComponent.kt :(. This is a known issue for GitHub: https://github.com/isaacs/github/issues/900, so it will possibly be fixed in the future. I  think the order we did this will be compatible with ``git --follow``. At least, I can confirm it works locally:

```
git log --oneline --stat -M --follow -- app/src/main/java/org/oppia/android/app/activity/ActivityComponent.kt
```

I might try the suggested Chrome extension after this PR is submitted to see if it's effective.

BenHenning commented:
CI checks are passing. Going ahead and merging to fix develop.

BenHenning commented:
Disabled the 'require branch to be up-to-date' check. Just sent out the 'all-clear' email.

BenHenning commented:
"Follow for Github" extension does not seem to work (see reviews--this might have to do with some GitHub redesigning).

The [Refined GitHub extension](https://github.com/sindresorhus/refined-github) seems to work, though:

Before
![image](https://user-images.githubusercontent.com/12983742/94201994-afa7f100-fe71-11ea-801c-233379c9e67f.png)

After
![image](https://user-images.githubusercontent.com/12983742/94202047-c8180b80-fe71-11ea-8501-a32b5762ffee.png)

I probably won't be using the extension yet, but it's a nice option to have.

BenHenning commented:
Conflicts when trying to CP into the release branch:

```
	both modified:   BUILD.bazel
	both modified:   WORKSPACE
	both modified:   app/src/main/AndroidManifest.xml
	both modified:   app/src/main/java/org/oppia/android/app/activity/ActivityComponent.kt
	both modified:   app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/administratorcontrols/administratorcontrolsitemviewmodel/AdministratorControlsDownloadPermissionsViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/application/ApplicationComponent.kt
	both modified:   app/src/main/java/org/oppia/android/app/application/ApplicationInjector.kt
	both modified:   app/src/main/java/org/oppia/android/app/application/ApplicationInjectorProvider.kt
	both modified:   app/src/main/java/org/oppia/android/app/completedstorylist/CompletedStoryListViewModel.kt
	deleted by us:   app/src/main/java/org/oppia/android/app/drawer/ExitProfileDialogFragment.kt
	deleted by us:   app/src/main/java/org/oppia/android/app/drawer/ExitProfileDialogInterface.kt
	both modified:   app/src/main/java/org/oppia/android/app/drawer/NavigationDrawerFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/help/faq/faqsingle/FAQSingleActivityPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/home/HomeActivity.kt
	both modified:   app/src/main/java/org/oppia/android/app/home/HomeFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/ongoingtopiclist/OngoingTopicListViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/options/OptionControlsViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/options/OptionsFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/player/audio/AudioFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivityPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/player/exploration/ExplorationManagerFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/player/exploration/HintsAndSolutionExplorationManagerFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/profile/PinPasswordViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/profile/ProfileChooserViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/profile/ResetPinDialogFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/profileprogress/ProfilePictureActivityPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/profileprogress/ProfileProgressViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditActivityPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/settings/profile/ProfileListViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/settings/profile/ProfileRenameActivityPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/settings/profile/ProfileResetPinActivityPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/splash/SplashActivityPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/story/StoryViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/topic/TopicViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/topic/info/TopicInfoFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/topic/practice/TopicPracticeViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/topic/questionplayer/HintsAndSolutionQuestionManagerFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/topic/revision/TopicRevisionViewModel.kt
	both modified:   app/src/main/java/org/oppia/android/app/walkthrough/end/WalkthroughFinalFragmentPresenter.kt
	both modified:   app/src/main/java/org/oppia/android/app/walkthrough/welcome/WalkthroughWelcomeFragmentPresenter.kt
	deleted by us:   app/src/main/res/layout-sw600dp/story_chapter_view.xml
	deleted by us:   app/src/main/res/layout-sw600dp/story_fragment.xml
	deleted by us:   app/src/main/res/layout-sw600dp/story_header_view.xml
	both modified:   app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/completedstorylist/CompletedStoryListActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/faq/FAQListFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/faq/FAQSingleActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/home/RecentlyPlayedFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/mydownloads/MyDownloadsFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/ongoingtopiclist/OngoingTopicListActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/options/OptionsFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/parser/HtmlParserTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/profile/AddProfileActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/profile/AdminAuthActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/profile/PinPasswordActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/profileprogress/ProfilePictureActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/profileprogress/ProfileProgressFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/recyclerview/BindableAdapterTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileListFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileResetPinActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/splash/SplashActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/story/StoryActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/testing/DragDropTestActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/testing/ImageRegionSelectionInteractionViewTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/testing/InputInteractionViewTestActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/testing/TestFontScaleConfigurationUtilActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/testing/TopicTestActivityForStoryTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/topic/TopicFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/topic/conceptcard/ConceptCardFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/topic/info/TopicInfoFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/topic/revision/TopicRevisionFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/utility/RatioExtensionsTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughFinalFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughTopicListFragmentTest.kt
	both modified:   app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughWelcomeFragmentTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/home/HomeActivityLocalTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/parser/StringToRatioParserTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/player/exploration/ExplorationActivityLocalTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/player/state/StateFragmentLocalTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/profile/ProfileChooserFragmentLocalTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/story/StoryActivityLocalTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/CompletedStoryListSpanTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/OngoingTopicListSpanTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/ProfileChooserSpanTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/ProfileProgressSpanCount.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/TopicRevisionSpanTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/options/DefaultAudioFragmentTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/options/OptionsFragmentTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/options/ReadingTextSizeFragmentTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/player/split/PlayerSplitScreenTesting.kt
	both modified:   app/src/test/java/org/oppia/android/app/testing/player/state/StateFragmentAccessibilityTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/topic/info/TopicInfoFragmentLocalTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentLocalTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityLocalTest.kt
	both modified:   app/src/test/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityLocalTest.kt
	both modified:   data/src/test/java/org/oppia/android/data/persistence/PersistentCacheStoreTest.kt
	both modified:   domain/BUILD.bazel
	both modified:   domain/src/main/java/org/oppia/android/domain/audio/CellularAudioDialogController.kt
	both modified:   domain/src/main/java/org/oppia/android/domain/exploration/ExplorationDataController.kt
	both modified:   domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt
	both modified:   domain/src/main/java/org/oppia/android/domain/onboarding/AppStartupStateController.kt
	both modified:   domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/AnalyticsController.kt
	both modified:   domain/src/main/java/org/oppia/android/domain/oppialogger/exceptions/ExceptionsController.kt
	deleted by us:   domain/src/main/java/org/oppia/android/domain/oppialogger/loguploader/LogUploadWorkManagerInitializer.kt
	deleted by us:   domain/src/main/java/org/oppia/android/domain/oppialogger/loguploader/LogUploadWorker.kt
	deleted by us:   domain/src/main/java/org/oppia/android/domain/oppialogger/loguploader/LogUploadWorkerFactory.kt
	deleted by us:   domain/src/main/java/org/oppia/android/domain/oppialogger/loguploader/LogUploadWorkerModule.kt
	deleted by us:   domain/src/main/java/org/oppia/android/domain/oppialogger/loguploader/WorkManagerConfigurationModule.kt
	both modified:   domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt
	both modified:   domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentProgressController.kt
	both modified:   domain/src/main/java/org/oppia/android/domain/question/QuestionTrainingController.kt
	both modified:   domain/src/main/java/org/oppia/android/domain/topic/StoryProgressController.kt
	both modified:   domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt
	both modified:   domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/audio/CellularAudioDialogControllerTest.kt
	deleted by us:   domain/src/test/java/org/oppia/android/domain/classify/rules/NumericInputIsInclusivelyBetweenRuleClassifierProviderTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/exploration/ExplorationDataControllerTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/exploration/ExplorationProgressControllerTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/onboarding/AppStartupStateControllerTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/oppialogger/analytics/AnalyticsControllerTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/oppialogger/exceptions/ExceptionsControllerTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/oppialogger/exceptions/UncaughtExceptionLoggerStartupListenerTest.kt
	deleted by us:   domain/src/test/java/org/oppia/android/domain/oppialogger/loguploader/LogUploadWorkManagerInitializerTest.kt
	deleted by us:   domain/src/test/java/org/oppia/android/domain/oppialogger/loguploader/LogUploadWorkerTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/profile/ProfileManagementControllerTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/question/QuestionAssessmentProgressControllerTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/question/QuestionTrainingControllerTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/topic/StoryProgressControllerTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/topic/StoryProgressTestHelperTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/topic/TopicControllerTest.kt
	both modified:   domain/src/test/java/org/oppia/android/domain/topic/TopicListControllerTest.kt
	deleted by us:   testing/src/main/java/org/oppia/android/testing/EditTextInputAction.kt
	both modified:   testing/src/main/java/org/oppia/android/testing/profile/ProfileTestHelper.kt
	both modified:   testing/src/test/java/org/oppia/android/testing/profile/ProfileTestHelperTest.kt
	both modified:   utility/BUILD.bazel
	deleted by us:   utility/src/main/java/org/oppia/android/util/data/DataProvidersInjector.kt
	deleted by us:   utility/src/main/java/org/oppia/android/util/data/DataProvidersInjectorProvider.kt
	deleted by us:   utility/src/main/java/org/oppia/android/util/logging/LogUploader.kt
	deleted by us:   utility/src/main/java/org/oppia/android/util/logging/firebase/FirebaseLogUploader.kt
	deleted by us:   utility/src/main/java/org/oppia/android/util/logging/firebase/FirebaseLogUploaderModule.kt
	both modified:   utility/src/test/java/org/oppia/android/util/data/DataProvidersTest.kt
```

BenHenning commented:
Due to the significant numbers of conflicts, this needs to be manually CP'd so that it can undergo code review.
